File: 3189-find-champion-ii/3189-find-champion-ii.cpp
class Solution {
public:
    int findChampion(int n, vector<vector<int>>& edges) {
        vector<int> vis(n, 0);
        for(int i = 0; i < edges.size(); i++){
            int u = edges[i][0];
            int v = edges[i][1];
            vis[v]++;
        }
        int res = -1, cnt = 0;
        for(int i = 0; i < n; i++){
            if(vis[i] == 0){
                if(cnt > 0){
                    return -1;
                }
                else{
                    cnt++;
                    res = i;
                }
            }
        }
        return res;
    }
};

File: 3189-find-champion-ii/3189-find-champion-ii.java
class Solution {
    public int findChampion(int n, int[][] edges) {
        int indeg[]=new int[n];
        for(int i=0;i<edges.length;i++){
            int a=edges[i][1];
            indeg[a]++;
        }

        int res=-1;
        int c=0;
        for(int i=0;i<n;i++)
              {
                if(indeg[i]==0){
                  if(c>0){
                    res=-1;
                     break;

                 }
        else{
            res=i;
            c++;
        }
    }

}   
return res;
 }
}